src/main/webapp/scripts/private-room-script.js

```diff
@@ -1,52 +1,198 @@
-// This code loads the IFrame Player API code asynchronously.
-var scriptTag = document.createElement('script');
-scriptTag.src = "https://www.youtube.com/iframe_api";
-var firstScriptTag = document.getElementsByTagName('script')[0];
-firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
-// Boolean used to handle statechange function
-var isVideoFinished = false;
+var roomId;
+var playlistUrls;
+var playlistIds;
 var youtubePlayer;
+var YT_BASE_URL = "https://www.youtube.com/embed/";
 
 /**
- * Fetches prvivate room and prints the room to console
+* Calls the three functions associated with loading the room's iframe
  */
-function fetchPrivateRoom() {
-  console.log('Fetching private room.');
-  // TODO: fetch hard-coded private room
+async function loadPlayerDiv(){
+    loadVideo();
+    // Get the room id from the private room's url
+    await getRoomId(window.location.href);
+    await fetchPrivateRoomVideo(roomId);
 }
 
 /**
-* This function creates an <iframe> (and YouTube player)
-* after the API code downloads.
-* TODO: Make videoId mutable
+* Get the room id by parsing the room's url
+* @param {url} String url of the current webpage
+* @return {roomId} room id at end of the url
+*/
+function getRoomId(url) {
+    var parser = document.createElement('a');
+    parser.href = url;
+    var query = parser.search.substring(1);
+    // Create a temp array of len=2, 
+    // holds 'roomId=' in [0] and room id in [1]
+    var tempArrayForRoomId = query.split('=');
+    var currentRoomId = tempArrayForRoomId[1];
+    roomId = currentRoomId;
+    return roomId;
+}
+
+/**
+* Take in currentRoomId and create a global array of youtube video urls and 
+* a global array of the video ids of the playlist
+* @param {currentRoomId} String Holds the room Id of the the user's room
+* @return {roomVideoUrl} The Url of the video to be displayed for the room
+ */
+async function fetchPrivateRoomVideo(currentRoomId) {
+    // Check that the current room id exits, then return playlist of given room
+    let roomPromise = await fetch('/verify-room?roomId='+roomId);
+    // fetch the json-version of the urls for all the youtube videos
+    let roomVideoUrl = await roomPromise.json();
+    // create an array of all the YT videos' urls
+    playlistUrls = parseJsonOfVideos(roomVideoUrl);
+    playlistIds = extractVideoIds(playlistUrls);
+}
+
+/*
+* Takes json of the urls of videos in playlist and puts them in a array of strings of urls
 */
+function parseJsonOfVideos(jsonOfVideos){
+    var playlistUrls = [];
+    for(i = 0; i < jsonOfVideos.length; i++) {
+        playlistUrls.push(jsonOfVideos[i]);
+    }
```

Comment:
I believe there is a function that copies the jsonOfVideos into platlistUrls

Reply:
Am i able to call that function from javascript?

Reply:
Looking at this function, it doesn't seem to do anything other than copy one list to another? All objects in javascript are json objects / arrays. How jsonOfVideos different to playlistUrls?

Reply:
Solved it!